Add WordPress Version Selection to Docker PHP Test Suite

adrianduffell opened the pull request:
This PR follows-up https://github.com/woocommerce/woocommerce-admin/pull/6119, adding the ability to select which WordPress version to use in the Docker PHP Test Suite.

```
WP_VERSION=5.6 npm run test:php 
```

WordPress 5.3 is used as the default version if none is selected. This is the [minimum supported version](https://github.com/woocommerce/woocommerce-admin/blob/8d04830d77f5bfdb6eb26089ff3b735d796dd2a6/readme.txt#L39).

The image version number tag was bumped to trigger an image rebuild automatically. This ensures the  ENTRYPOINT changes activate on next use.

The README documentation was also updated.

### Detailed test instructions:

-   Run the test suite normally: `npm run test:php`
-   In the output, confirm the image is rebuilt.

```
Building phpunit
Step 1/12 : FROM alpine
 ```

-   In the output, confirm the installation script is triggered and WordPress 5.3 is used by default:

```
+ curl -s https://wordpress.org/wordpress-5.3.6.tar.gz
```
```
+ svn co --quiet https://develop.svn.wordpress.org/branches/5.3/tests/phpunit/includes/ /tmp/wordpress-tests-lib/includes
+ svn co --quiet https://develop.svn.wordpress.org/branches/5.3/tests/phpunit/data/ /tmp/wordpress-tests-lib/data
```

-   Run the test suite against the latest WordPress version: `WP_VERSION=5.6 npm run test:php`

-   In the output, confirm the installation script is triggered and WordPress 5.6 is used:

```
+ curl -s https://wordpress.org/wordpress-5.6.1.tar.gz
```
```
+ svn co --quiet https://develop.svn.wordpress.org/branches/5.5/tests/phpunit/includes/ /tmp/wordpress-tests-lib/includes
+ svn co --quiet https://develop.svn.wordpress.org/branches/5.5/tests/phpunit/data/ /tmp/wordpress-tests-lib/data
```

-   Run the test suite against the latest WordPress again: `WP_VERSION=5.6 npm run test:php`

-   Confirm the installation script is **NOT** triggered. PHPUnit should begin immediately.

adrianduffell committed:
Add WordPress version selection to Docker PHP test suite

samueljseay reviewed (commented):
This is looking good, just a couple things:

1. I noticed this myself when doing the gh actions migration that the wp install script does a WP version check that just doesn't seem to work (I wonder if it only works with versions like x.x.x). I'm happy to pre approve you removing this condition from the script because it doesn't really add any value right now. the error in the output: `bin/install-wp-tests.sh: line 22: [: 5.3: integer expression expected`

2. I get a fatal error trying to run with WP_VERSION=5.6:

```
+ cd /app
PHP Warning:  require_once(/tmp/wordpress/wp-includes/PHPMailer/PHPMailer.php): failed to open stream: No such file or directory in /tmp/wordpress-tests-lib/includes/mock-mailer.php on line 2

Warning: require_once(/tmp/wordpress/wp-includes/PHPMailer/PHPMailer.php): failed to open stream: No such file or directory in /tmp/wordpress-tests-lib/includes/mock-mailer.php on line 2
PHP Fatal error:  require_once(): Failed opening required '/tmp/wordpress/wp-includes/PHPMailer/PHPMailer.php' (include_path='.:/usr/share/php7') in /tmp/wordpress-tests-lib/includes/mock-mailer.php on line 2

Fatal error: require_once(): Failed opening required '/tmp/wordpress/wp-includes/PHPMailer/PHPMailer.php' (include_path='.:/usr/share/php7') in /tmp/wordpress-tests-lib/includes/mock-mailer.php on line 2
```

adrianduffell commented:
> I get a fatal error trying to run with WP_VERSION=5.6:

@samueljseay This seems like WordPress didn't install properly 😕  Do you still have the full output from the install script?

adrianduffell committed:
Remove guard for old WordPress version

It was producing an error message when valid versions like “5.3” are selected:

bin/install-wp-tests.sh: line 22: [: 5.3: integer expression expected

I removed it since disallowing older versions doesn't really add any value now.

adrianduffell commented:
> the wp install script does a WP version check that just doesn't seem to work

@samueljseay Thanks, I've been wondering about this error message too. Good idea to remove the condition from the script. I've done that in 98d6f58

adrianduffell committed:
Bump image version to trigger rebuild

adrianduffell committed:
Fix existing WordPress version not being replaced

The install-wp-tests.sh script does not install WordPress if a previous installation is detected.

To enable WP version switching, the ENTRYPOINT removes previous files to ensure the installation starts from a clean state.

adrianduffell commented:
I was able to reproduce the fatal error after testing it from a clean slate.

I didn’t realize the install script [keeps the existing installation of WordPress if it exists](https://github.com/woocommerce/woocommerce-admin/blob/a1aae58a19d942468e9dc24d571f2610cba28cc5/bin/install-wp-tests.sh#L63-L65). It does however[ replace the WP test files](https://github.com/woocommerce/woocommerce-admin/blob/a1aae58a19d942468e9dc24d571f2610cba28cc5/bin/install-wp-tests.sh#L111-L112). As such, when `WP_VERSION=5.6` was selected, it downloaded the 5.6 test files but continued using WordPress 5.3 😅 

I’ve fixed it in 35f86f9 by ensuring any previous installations are removed.

adrianduffell committed:
Partially revert abd8ab80dc7ef6b57316636d467b511a07d0c5d3

The typo fix was accidentally included in this commit.

adrianduffell committed:
Fix installation routine running every time

A typo was causing the installation routine to run on each use.

adrianduffell commented:
I've updated the testing instructions about checking the correct version of WordPress is installed.

Also fixed a typo in 37dfef6.

@samueljseay Could you take a look at this again? Thanks!

samueljseay reviewed (approved):
This is working well for me now. Thanks for this!
